### PR TITLE
Auto-detecting node IP, looking for interface named `InterfaceName`

### PR DESCRIPTION
**Pull request by xmapst**
**Reason for PR**:

Append "--node-ip" param in kubelet startup params. default Auto-detecting  node IP, User can use
"InterfaceName" parameter variable to configure the node IP used by kubelet.

**Issue Fixed**:

Issue #86 

**Requirements**

- [ ] Sqaush commits 
- [ ] Documentation
- [x] Tests

**Notes**:

### CODE REVIEW

**Comment by xmapst** [comment]
@benmoss @ddebroy Please merge

**Comment by xmapst** [comment]
@ddebroy @benmoss

**Comment by xmapst** [comment]
image

File: kubeadm/scripts/PrepareNode.ps1 [hunk 1]
```diff
@@ -104,10 +115,11 @@ if ($global:containerRuntime -eq "Docker") {
     }
 }
 
-$cmd = "C:\k\kubelet.exe $global:KubeletArgs --cert-dir=$env:SYSTEMDRIVE\var\lib\kubelet\pki --config=/var/lib/kubelet/config.yaml --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf --hostname-override=$(hostname) --pod-infra-container-image=`"mcr.microsoft.com/oss/kubernetes/pause:1.4.1`" --enable-debugging-handlers --cgroups-per-qos=false --enforce-node-allocatable=`"`" --network-plugin=cni --resolv-conf=`"`" --log-dir=/var/log/kubelet --logtostderr=false --image-pull-progress-deadline=20m"
+$cmd = "C:\k\kubelet.exe $global:KubeletArgs --cert-dir=$env:SYSTEMDRIVE\var\lib\kubelet\pki --config=/var/lib/kubelet/config.yaml --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf --hostname-override=$(hostname) --node-ip=`"{{NODE_IP}}`" --pod-infra-container-image=`"mcr.microsoft.com/oss/kubernetes/pause:1.4.1`" --enable-debugging-handlers --cgroups-per-qos=false --enforce-node-allocatable=`"`" --network-plugin=cni --resolv-conf=`"`" --log-dir=/var/log/kubelet --logtostderr=false --image-pull-progress-deadline=20m"
```
Comment (jsturtevant) [inline kubeadm/scripts/PrepareNode.ps1:118]: Is there a reason we need to have this specified here and not use the optional flags for env specific flags as outlined in: 

https://github.com/kubernetes-sigs/sig-windows-tools/issues/86#issuecomment-628343619 

> It seems that kubelet will retrive params from file kubeadm-flags.env at startup. So I think user can specify the --node-ip in this file for the NodeIP config.
>
> By this way we don't need to change the PrepareNode.ps1 or StartKubelet.ps1 for this config.

**Pull request closed without merging**